### PR TITLE
CAMT-604 Implement flash feature for capturing photo

### PR DESCRIPTION
**Pull request by saurabhathavale**
implemented for photo capture only
will create different PR for video capture

### CODE REVIEW

File: app/src/main/java/com/leia/leiacam2/CameraFragment.kt [hunk 1]
```diff
@@ -82,17 +83,34 @@ class CameraFragment : Fragment(), CaptureInterface, SwitchInterface {
         mCameraUIBinding!!.captureTypeUI.setViewModel(this, mViewModel)
     }
 
+    private fun createObserver() {
```
Comment (Sean7852) [inline app/src/main/java/com/leia/leiacam2/CameraFragment.kt:86]: Create what observer? Please optimize the name

File: app/src/main/java/com/leia/leiacam2/camera/ImageCapturing.kt [hunk 2]
```diff
@@ -23,9 +21,18 @@ interface ImageCapturingListener {
...
 
-    fun takePictureAsync(lifecycleCoroutineScope: LifecycleCoroutineScope, captureSession: CameraCaptureSession, imageReader: ImageReader, width: Int, height: Int, listener: ImageCapturingListener) {
+    fun takePictureAsync(
+        lifecycleCoroutineScope: LifecycleCoroutineScope,
+        captureSession: CameraCaptureSession,
+        imageReader: ImageReader,
+        width: Int,
+        height: Int,
+        listener: ImageCapturingListener,
+        viewModel: CameraViewModel
```
Comment (Sean7852) [inline app/src/main/java/com/leia/leiacam2/camera/ImageCapturing.kt:31]: Can we pass in the `FlashMode` directly? `FlashMode` has smaller scope compared to `CameraViewModel`,  pass in the minimal reference is always the first choice.

Comment (saurabhathavale) [reply app/src/main/java/com/leia/leiacam2/camera/ImageCapturing.kt:31]: it that case we may need to pass in other parameters in the future like hdr ,mfnr as a variable.

File: app/src/main/java/com/leia/leiacam2/CameraFragment.kt (hunk 1 above)
Comment (saurabhathavale) [reply app/src/main/java/com/leia/leiacam2/CameraFragment.kt]: done

File: app/src/main/java/com/leia/leiacam2/camera/LeiaCameraManager.kt [hunk 3]
```diff
@@ -62,7 +65,7 @@ class LeiaCameraManager(
 
     fun captureImage(lifecycleCoroutineScope: LifecycleCoroutineScope, onImageTaken: (byteArray: ByteArray) -> Unit) {
```
[the remark this answers is not in the document]
Comment (saurabhathavale) [reply app/src/main/java/com/leia/leiacam2/camera/LeiaCameraManager.kt]: done

File: app/src/main/java/com/leia/leiacam2/camera/ImageCapturing.kt (hunk 2 above)
Comment (saurabhathavale) [reply app/src/main/java/com/leia/leiacam2/camera/ImageCapturing.kt]: add a interface to get flash mode

**Review by Sean7852 — APPROVED** [review]

**Pull request merged**